wgsl/index.bs

```diff
@@ -689,84 +682,259 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
-    OpenGL `std140` layout is the same as using the tightest offset and stride assignments in [SHORTNAME] [=uniform buffer layout=].
-* OpenGL `std140` layout of `mat2x2` has extra padding between column vectors that is not present in a `mat2x2` type in [SHORTNAME].
-    * The OpenGL `std140` layout of the `mat2x2` type has the second column vector starting 16 bytes after the first column vector.
-        But in [SHORTNAME] the second column vector of `mat2x2<f32>` starts 8 bytes after the first column vector.
-* For any type, OpenGL `std430` layout is the same as using the tightest offset and stride assignments in [SHORTNAME] [=storage buffer layout=].
-* OpenGL supports row-major matrices, but [SHORTNAME] does not.
-
-Compared to [[Vulkan1.2ext|Vulkan &sect; 15.6.4 Offset and Stride Assignment]]:
-* Vulkan *standard buffer layout* maps to [SHORTNAME] [=standard buffer layout rules=] with the following qualifications:
-    * Vulkan allows a vector to be aligned to the size of its scalar component, but [SHORTNAME] requires a more constrained alignment.
-    * The Vulkan `scalarBlockLayout` and `uniformBufferStandardLayout` features do not apply to [SHORTNAME].
-    * The Vulkan concept of *scalar alignment* does not correspond to a concept in [SHORTNAME];
-* The Vulkan *base alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the [=storage classes/storage=] storage class: |Align|(|S|,`storage`).
-* The Vulkan *extended alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the [=storage classes/uniform=] storage class: |Align|(|S|,`uniform`).
-* The Vulkan concept of *improperly straddle* is not permitted in [SHORTNAME], because [SHORTNAME] requires vectors to be aligned to their whole size.
-* Vulkan supports non-32-bit scalar types and vector types with non-32-bit components, but [SHORTNAME] does not.
-* Vulkan supports row-major matrices, but [SHORTNAME] does not.
-* Vulkan allows offsets to be non-monotonic, but [SHORTNAME] does not.
-
-#### Internal Layout of Values #### {#internal-value-layout}
+Structure members will use their type's size and alignment, unless the
+structure member is explicitly annotated with [=size=] and / or [=align=]
+decorations, in which case those member decorations take precedence.
+
+The first structure member always has a zero byte offset from the start of the
+structure.
+
+Subsequent members have the following byte offset from the start of the structure:
+<p algorithm="structure member offset">
+  [=OffsetOf=](|S|, M<sub>N</sub>) = [=roundUp=]([=AlignOf=](|S|, M<sub>N</sub>), [=OffsetOf=](|S|, M<sub>N-1</sub>) + [=SizeOf=](|S|, M<sub>N-1</sub>)<br>
+  Where M<sub>N</sub> is the current member and M<sub>N-1</sub> is the previous member
+</p>
+
+Structure members must not overlap. Structure members decorated with the
+[=size=] attribute must have a size at least as large as the size of the
+member's type.
```

Comment:
I always stumble when I read this because of course the size is as large as the size. I specified it after all.

```suggestion
Structure members must not overlap. If a structure member is decorated with the [=size=] attribute, the value must be at least as large as the default size of the member's type.
```

Reply:
Done